v0/destinations/airship/transform.js

```diff
@@ -0,0 +1,119 @@
+const btoa = require("btoa");
+const { EventType } = require("../../../constants");
+
+const {
+  CONFIG_CATEGORIES,
+  MAPPING_CONFIG,
+  BASE_URL_EU,
+  BASE_URL_US
+} = require("./config");
+
+const {
+  constructPayload,
+  defaultPostRequestConfig,
+  removeUndefinedAndNullValues,
+  defaultRequestConfig,
+  CustomError
+} = require("../../util");
+
+function responseBuilderSimple(message, category, destination) {
+  const payload = constructPayload(message, MAPPING_CONFIG[category.name]);
+  const { appKey, dataCenter, appSecret } = destination.Config;
+  const response = defaultRequestConfig();
+  let BASE_URL = BASE_URL_US;
+  // check the region and which api end point should be used
+  if (dataCenter) {
+    BASE_URL = BASE_URL_EU;
+  }
+
+  if (
+    message.type === "identify" &&
+    message.context.traits.email &&
+    message.traits.channelId
+  ) {
+    response.endpoint = `${BASE_URL} /api/named_users/associate`;
+  }
+  if (message.context.traits.remove || message.context.traits.add) {
+    response.endpoint = `${BASE_URL} /api/named_users/tags`;
+  }
+  if (payload.attributes) {
+    response.endpoint = `${BASE_URL} /api/named_users/${message.traits.user_id}/attributes`;
+  }
+  if (message.type === "track") {
+    response.endpoint = `${BASE_URL} /api/custom-events`;
+  }
+  response.method = defaultPostRequestConfig.requestMethod;
+  response.headers = {
+    "Content-Type": "application/json",
+    Accept: "application/vnd.urbanairship+json; version=3",
+    Authorization: `Basic ${btoa(`${appKey}:${appSecret}`)}`
+  };
+  const lowerCaseName = message.context.traits.toLowerCase();
```

Comment:
traits is an object. You can't apply `toLowerCase()` to an object.

Reply:
resolved